main/ahb_for_sles.go

```diff
@@ -125,6 +125,12 @@ func _hasPubCloudMod(pubCloudService string) (bool, string, string, error) {
 					}
 				}
 				if distro != "" {
+					if strings.Contains(value["repo_1"], distro+"-SP") {
+						// get SP if any
+						indexSp := strings.LastIndex(value["repo_1"], distro+"-SP")
+						sp := value["repo_1"][(indexSp + 5):(indexSp + 6)]
+						distro = distro + "." + sp
+					}
```

Comment:
I missed this previously, obviously. But the function `hasPubCloudMod` should return, based on it's name true or false depending on if we have the public cloud module or not on the system. For the distro information we should have a separate function and read from `/etc/os-release` and fo rthe architecture there should be a built in Go function to call.